ci/e2e-tests/test-run.sh

```diff
@@ -54,12 +54,27 @@ get_package() {
     #
     # Ref: https://docs.github.com/en/rest/reference/actions#list-workflow-runs
     echo "Getting latest workflow run's artifacts URL..." >&2
-    artifacts_url="$(
```

Comment:
You haven't changed the `set -euo pipefail` at the start of the script, so any command that fails will still abort the script entirely.

You should still leave the `set` since it catches general programming errors, but it means that any command that you want to retry needs to suppress that, either by appending `|| :` or doing `set +e` for that command.

Reply:
I think the command still returns 0 (success), but the result is HTTP 429 and the jquery parse is 'null'. So it should be fine if e is still set.

Reply:
And if curl does fail and returns a non-zero exit code? The point of this change was to retry when these external operations fail, right?

Reply:
Ah right. Fixed.

Reply:
This is still broken if curl succeeds but outputs something that isn't JSON, because jq will choke on that.

I think the cleanest solution, just like the other place, is to do a `set +e` inside the subshell used to run `github_curl | jq` and then check `$artifacts_url` for both `''` and `'null'`

Reply:
Oh right. Updated.